app/src/main/java/com/mashup/ui/widget/MashUpButton.kt

```diff
@@ -0,0 +1,182 @@
+package com.mashup.ui.widget
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.core.animateFloat
+import androidx.compose.animation.core.infiniteRepeatable
+import androidx.compose.animation.core.keyframes
+import androidx.compose.animation.core.rememberInfiniteTransition
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.geometry.center
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.StrokeCap
+import androidx.compose.ui.graphics.drawscope.Stroke
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.mashup.core.ui.colors.*
+import com.mashup.core.ui.theme.MashUpTheme
+import com.mashup.core.ui.typography.Body1
+
+enum class ButtonStyle(val backgroundColor: Color, val textColor: Color) {
+    PRIMARY(backgroundColor = Brand500, textColor = Color.White),
+    DISABLE(backgroundColor = Brand300, textColor = Color.White),
+    DEFAULT(backgroundColor = Gray100, textColor = Gray600)
+}
+
+@Composable
+fun MashUpButton(
+    modifier: Modifier = Modifier,
+    buttonStyle: ButtonStyle = ButtonStyle.PRIMARY,
+    text: String,
+    onClick: () -> Unit,
+    isEnabled: Boolean = true,
+    showLoading: Boolean = false,
+) {
+    Box(
+        modifier = modifier
+            .clip(RoundedCornerShape(12.dp))
+            .fillMaxWidth()
+            .height(52.dp)
+            .background(if (isEnabled) buttonStyle.backgroundColor else ButtonStyle.DISABLE.backgroundColor)
+            .clickable(
+                indication = null,
+                interactionSource = remember { MutableInteractionSource() }
+            ) { if (isEnabled) onClick() },
```

Comment:
이렇게 enabled 속성을 사용하는건 어떤가요?? 👀 
아래처럼 바꾸면 onClick 매개변수를 바로 넘길 수 있을 것 같아요!

```suggestion
         .clickable(
                indication = null,
                enabled = isEnabled,
                interactionSource = remember { MutableInteractionSource() },
                onClick = onClick
            ),
```

Reply:
굿굿 위에 코드리뷰랑 같이 수정했어요~

2fcaa3e686f5eacc8f649d764211af2f65f5ab0f [🛠 MashUpButton 매개변수 재정비](https://github.com/mash-up-kr/mashup_Android/pull/192/commits/2fcaa3e686f5eacc8f649d764211af2f65f5ab0f)

Reply:
ㅋㅋㅋ재정비..
enabled == false면 onClick도 수행 안되는거죠??

Reply:
ㅋㅎ.. 넹 마자여!